src/routes/safe/components/Settings/UpdateSafeModal/index.jsx

```diff
@@ -0,0 +1,115 @@
+// @flow
+import React from 'react'
+import IconButton from '@material-ui/core/IconButton'
+import Close from '@material-ui/icons/Close'
+import { withStyles } from '@material-ui/styles'
+import Row from '~/components/layout/Row'
+import Paragraph from '~/components/layout/Paragraph'
+import Hairline from '~/components/layout/Hairline'
+import GnoForm from '~/components/forms/GnoForm'
+import Block from '~/components/layout/Block'
+import Button from '~/components/layout/Button'
+import { styles } from './style'
+import { DELEGATE_CALL } from '~/logic/safe/transactions'
+import { getEncodedMultiSendCallData, getGnosisSafeInstanceAt, multiSendAddress } from '~/logic/contracts/safeContracts'
+import type { MultiSendTransactionInstanceType } from '~/logic/contracts/safeContracts'
+
+
+type Props = {
+  onClose: Function,
+  createTransaction: Function,
+  classes: Object,
+  safeAddress: string,
+}
+
+
+const UpdateSafeModal = ({
+  onClose, classes, safeAddress, createTransaction,
+}: Props) => {
+  const handleSubmit = async () => {
+    // Call the update safe method
+    const sendTransactions = async (txs: Array<MultiSendTransactionInstanceType>) => {
+      const encodeMultiSendCallData = getEncodedMultiSendCallData(txs)
+      createTransaction({
+        safeAddress,
+        to: multiSendAddress,
+        valueInWei: 0,
+        txData: encodeMultiSendCallData,
+        notifiedTransaction: 'STANDARD_TX',
+        enqueueSnackbar: () => {},
+        closeSnackbar: () => {},
+        operation: DELEGATE_CALL,
+      })
+    }
+    const safeInstance = await getGnosisSafeInstanceAt(safeAddress)
+    const fallbackHandlerTxData = safeInstance.contract.methods.setFallbackHandler('0xd5D82B6aDDc9027B22dCA772Aa68D5d74cdBdF44').encodeABI()
+    const updateSafeTxData = safeInstance.contract.methods.changeMasterCopy('0x34CfAC646f301356fAa8B21e94227e3583Fe3F5F').encodeABI()
+    const txs = [
+      {
+        operation: 0,
+        to: safeAddress,
+        value: 0,
+        data: fallbackHandlerTxData,
+      },
+      {
+        operation: 0,
+        to: safeAddress,
+        value: 0,
+        data: updateSafeTxData,
+      },
+    ]
+
+
+    await sendTransactions(txs)
```

Comment:
can we move all this to an external/helper function? Like `upgradeSafe` or something around that

Reply:
done